src/dtx/dtx_common.c

```diff
@@ -297,6 +297,14 @@ dtx_handle_init(struct dtx_id *dti, daos_handle_t coh, struct dtx_epoch *epoch,
 	dth->dth_dti_cos_count = dti_cos_cnt;
 	dth->dth_ent = NULL;
 	dth->dth_flags = leader ? DTE_LEADER : 0;
+
+	/* Set 'DTE_BLOCK' flag for EC object modification or
+	 * distributed transaction.
+	 */
+	if (daos_oclass_is_ec(leader_oid->id_pub, NULL) ||
+	    (mbs != NULL && mbs->dm_grp_cnt > 1))
+		dth->dth_flags |= DTE_BLOCK;
+
```

Comment:
just to confirm that seems did not see where in this patch do the sync commit when the DTE_BLOCK is set?
and do we still need the dth->dth_sync flag after this patch?

Reply:
Yes, dth_sync is still needed. It contains not only the cases of the DTX with "DTE_BLOCK" flag, but also some other cases, such as fail to add the async mode DTX into CoS cache.